src/org/labkey/remoteapi/issues/IssueModel.java

```diff
@@ -0,0 +1,213 @@
+package org.labkey.remoteapi.issues;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.labkey.api.collections.CaseInsensitiveHashMap;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class IssueModel
+{
+    // keys
+    private final String TITLE= "title";
+    private final String ISSUE_ID = "issueid";
+    private final String ISSUE_DEF_NAME = "issueDefName";
+    private final String ASSIGNED_TO = "assignedto";
+    private final String ACTION = "action";
+    private final String TYPE = "type";
+    private final String PRIORITY = "priority";
+    private final String COMMENT = "comment";
+    private final String NOTIFY_LIST = "notifyList";
+    private final String RESOLUTION = "resolution";
+    private final String RESOLVED = "resolved";
+    private final String STATUS = "status";
+
+    private final Map<String, Object> _properties = new CaseInsensitiveHashMap<>();
+    private final List<File> _attachments = new ArrayList();
+
+    // https://www.labkey.org/Documentation/wiki-page.view?name=sampleJSscripts#issues
+    public IssueModel()
+    {
+    }
+
+    public IssueModel setProperties(Map properties)
+    {
+        _properties.putAll(properties);
+        return this;
+    }
+
+    public JSONObject toJSON()
+    {
+        var json = new JSONObject(_properties);
+
+        // handle attachments
+        if (!_attachments.isEmpty())
+        {
+            List<String> names = _attachments.stream()
+                    .map(f -> f.getName())
+                    .collect(Collectors.toList());
+
+            json.put("attachment", String.join("|", names));
+        }
+
+        return json;
+    }
+
+    public IssueAction getAction()
+    {
+        return IssueAction.valueOf(_properties.get(ACTION).toString());
+    }
+
+    public IssueModel setAction(IssueAction action)
+    {
+        _properties.put(ACTION, action.getValue());
+        return this;
+    }
+
+    public Long getIssueId()
+    {
+        if (_properties.get(ISSUE_ID) != null)
+            return (Long)_properties.get(ISSUE_ID);
+        else
+            return null;
+    }
+
+    public IssueModel setIssueId(Long issueId)
+    {
+        _properties.put(ISSUE_ID, issueId);
+        return this;
+    }
+
+    public String getTitle()
+    {
+        return (String) _properties.get(TITLE);
+    }
+
+    public IssueModel setTitle(String title)
+    {
+        _properties.put(TITLE, title);
+        return this;
+    }
+
+    // per https://www.labkey.org/Documentation/wiki-page.view?name=sampleJSscripts#issues, for a single notify user
+    // you can supply an email address as a string; for multiple notifies, userIDs are necessary
+    public String getNotify()
+    {
+        return (String) _properties.get("notifyList");
```

Comment:
Should this use the constant used by the setter?
```suggestion
        return (String) _properties.get(NOTIFY_LIST);
```

Reply:
Yes, it should- these keys now use enum values tho